client-extensions/liferay-object-collection-filter-extension/src/page/Search.js

```diff
@@ -0,0 +1,104 @@
+import React,  { useState, useEffect, Suspense } from 'react';
+import ClayLayout from '@clayui/layout';
+import SearchResult from '../common/components/SearchResult.js';
+import SearchInput from '../common/components/SearchInput.js';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import {ClayPaginationBarWithBasicItems} from '@clayui/pagination-bar';
+import { findObject } from '../common/services/liferay/object.js';
+import FilterSettings from "../common/components/FilterSettings";
+import FilterForm from "../common/components/FilterForm";
+
+const Search = () => {
+	const [searchKeyword, setSearchKeyword] = useState('');
+	const [delta, setDelta] = useState(20);
+	const [page, setPage] = useState(1);
+	const [error, setError] = useState(null);
+	const [filters, setFilters] = useState([{
+		type: 'multiple-picklist',
+		property: 'multiplePicklistField',
+		values: [
+				"item1",
+				"item2"
+		]
+	}, {
+			type: 'numeric',
+			property: 'abc',
+			values: [0, 50]
+	}]);
+	const [searchResults, setSearchResults] = useState(null);
+	const [url, setURL] = useState('o/c/myobjects/')
+	const [total, setTotal] =  useState(1);
+	const [showSettings, setShowSettings] = useState(false);
+	const [settings, setSettings] = useState({
+		selectedObject: '',
+		selectedFields: [],
+	});
+
+	const handleSettingsClose = (newSettings) => {
+		setShowSettings(false);
+		setSettings((prevSettings) => ({
+			...prevSettings,
+			...newSettings,
+		}));
+	};
+
+	const handleSearchKeywordChange = (event) => {
+		setSearchKeyword(event.target.value);
+	};
+
+	const handleSearch = async () => {
+		try {
+			const data = await findObject(url, searchKeyword, filters, page, delta);
+
+			setSearchResults(data.items);
+			setTotal(data.totalCount)
+			setError(null);
+		} catch (error) {
+			setError(error.message);
+		}
+	};
+
+	useEffect(() => {
+		handleSearch()
+	},[delta, page])
+
+	return (
+		<div>
+			{Liferay.ThemeDisplay.isSignedIn() && (
+				<Suspense fallback={<ClayLoadingIndicator displayType="secondary" size="sm" />}>
+					<ClayLayout.Container view>
+						<SearchInput handleSearchChange={handleSearchKeywordChange} handleSearchSubmit={handleSearch}/>
+						<button onClick={() => setShowSettings(true)}>Settings</button>
+						{showSettings &&
+							<FilterSettings
+								initialSettings={settings}
+								onSaveAndClose={handleSettingsClose}
+							/>
+						}
+						<FilterForm selectedFields={settings.selectedFields}/>
+						{error && <p>Error: {error}</p>}
+						{searchResults && searchResults.length == 0 && <p>No results found </p>}
```

Comment:
it should be 
`{searchResults?.length === 0 ? <p>No results found</p> : null}
{searchResults?.length > 0 ? <SearchResult items={searchResults} /> : null}`